dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -211,6 +233,22 @@ def is_match(self) -> bool:
             is_match = True
         return is_match
 
+    def update_stop_condition(self, data: DataFrame):
+        """Return value stop_condition_is_met given stop conditions.
+
+        :param data: Dataframe currently being processed by categorical profiler
+        :type data: DataFrame
+        :return: boolean for stop conditions
+        """
+        if (
+            self.max_sample_size_to_check_stop_condition is not None
+            and len(data) >= self.max_sample_size_to_check_stop_condition
+            and self.stop_condition_unique_value_ratio is not None
+            and len(self._categories) / len(data)
```

Comment:
```python
merged_unique_count = len(self._categories)
merged_sample_size = (self.sample_size + len(data))
merged_unique_ratio =  merged_unique_count / merged_sample_size


if (
    self.max_sample_size_to_check_stop_condition is not None
    and  self. stop_condition_unique_value_ratio is not None
    and merged_sample_size >= self.max_sample_size_to_check_stop_condition
    and merged_unique_ratio >= stop_condition_unique_value_ratio
):
    self._stop_condition_is_met = True
    self._stopped_at_unique_ratio = merged_unique_ratio
    self._stopped_at_unique_count = merged_unique_count
```
I think this keeps it clear and then sets the values.

An option would be if we want this function to  just be a check, we can calculate in`_update_categories `:
merged_unique_count = len(self._categories)
merged_sample_size = (self.sample_size + len(data))
merged_unique_ratio =  merged_unique_count / merged_sample_size
```
and pass it into the check function:
```python
def _check_stop_condition_is_met(self, sample_count, sample_size, unqiue_ratio):
    """Return value stop_condition_is_met given stop conditions.
    
    :return: boolean for stop conditions
    """
    if (
          self.max_sample_size_to_check_stop_condition is not None
          and  self. stop_condition_unique_value_ratio is not None
          and sample_size >= self.max_sample_size_to_check_stop_condition
          and unique_ratio >= stop_condition_unique_value_ratio
    ):
        return True
    return False
```

then in `_update_categories`:
```python
if self._check_stop_condition_is_met(merged.....):
    self._stop_condition_is_met = True
    self._categories = {}
    self._stopped_at_unique_ratio = merged_unique_ratio
    self._stopped_at_unique_count = merged_unique_count
```
^^ similar logic can then be used in the `__add__` where we calc the:
```python
if not profile1._stop_condition_is_met and not profile2._stop_condition_is_met:
    # merge dicts and then check the values for the merging with the stop condition, etc.
else:
    # merged categories is empty
    # unique ratio and unique count can be take from the profile with the largest sample size
    # if we avg, will be problem for streams.
```
between the two profies